---
title: "Differential Expression Analysis, FGSEA Analysis, and Discussion Questions"
author: "Yuki Ito"
output: html_document
---


## Evaluate the QC metrics for the full data



# Project 1 - Discussion Questions
These questions are meant to help me assess how well I’ve conveyed some of the
material as well as challenge you to think conceptually about some of the
underlying science. Some of the questions have a definitive "right" answer while
some of them are more open-ended and I'm just looking for your thought process
and interpretation. All of these questions can be answered in a few sentences so
do your best to be concise.

Please copy these questions into your single Rmd or notebook that you've been 
working on your DE analysis. All of your analyses, writing, and answers to these
questions should be in a single notebook. 


## RNAseq

1. List the major high-level steps of a basic RNAseq experiment to look for
differentially expressed genes. At each step, list what data you need to perform
each step and what format they are in (if applicable). At minimum, there are 4
essential steps.







2. Consider the following FastQC plot.

```{r}
knitr::include_graphics("fastqc_plot.png")
```


2a. What aspect of the data does this plot show?







2b. Make an interpretation of this plot assuming the data type was RNASeq.







2c. Do you think this plot indicates there was a problem with the dataset?
Explain your answer.






2d. Make a hypothesis about what the problem represents. Outline a bioinformatics
strategy that would enable you to confirm your hypothesis. 






3. What is a splice-aware aligner? When is it important to use a splice-aware
aligner?







4. What does a “gene-level” count as produced by VERSE or any other counting
tool in a RNAseq experiment represent?






5. In your own words, briefly describe what information the matching GTF for a
reference genome stores.






6. When counting alignments using VERSE or any other utility, why do we need to
provide the matching reference genome GTF file?







7. Let’s pretend that this was a GSEA result from an experiment where we treated
293T cells with a drug and compared changes in gene expression to wild-type
cells treated with a vehicle control. The differential expression results are
relative to the control cells (i.e. a positive fold change means a gene is
upregulated upon treatment with the drug)

Assume the following result is statistically significant with a positive NES
(normalized enrichment score) and that it represents a GSEA experiment performed
on the entire list of genes discovered in the experiment ranked by
log2FoldChange (i.e. genes that are “upregulated” in the cells treated with drug
are on the “left” and genes that are “downregulated” due to treatment are on the
“right”).

```{r}
knitr::include_graphics("gsea_plot.png")
```


7a. Form a valid interpretation / conclusion of the results shown in the plot
given the above setup.





7b. Now consider that all of the genes driving the enrichment are all activators
of the inflammatory pathway. Does your interpretation change and if so, how?





7c. Finally, consider that all of the genes driving the enrichment all function
to inhibit inflammation. Does your interpretation change and if so, how?






8. Rank the following quality control metrics for a 2x100nt paired-end illumina 
mRNAseq dataset from most concerning to least concerning. Provide a brief
statement on where you ranked each and why. Assume that the data has not been
processed and was provided as-is from the sequencing machine. 

- Unequal Read Lengths
- Average PHRED score < 20 in the last 10 bases
- 15% of reads have identical sequences
- 50% of reads are multimapped after alignment to the appropriate genome
- 10% of reads are unmapped after alignment to the appropriate genome
- Non-random nucleotide distribution in the first 6 bases
- Nucleotide frequencies of ACTG are not equal over the entire read
- Unequal number of forward and reverse reads